docs/backstage-guide-v1.md

```diff
@@ -0,0 +1,313 @@
+* **Integration:** Seamlessly connect with external tools and services.
+* **Automation:** Automate repetitive tasks to streamline workflows
+
+### Steps to Create a Custom Backstage Plugin
+
+**Step 1: Create a Backstage Plugin**
+
+From the root of your project, use the following command to create a new plugin:
+
+```bash
+yarn new --select plugin
+```
+- This command will invoke the Backstage CLI to generate a new plugin based on the provided ID (example: simple-plugin). 
+- The plugin will be automatically built and added to your Backstage App.
+- If your Backstage App is already running (via yarn start or yarn dev), you can view the default page for your new plugin by navigating to: `http://localhost:3000/simple-plugin`
+
+**Step 2: Simplifying to a "Hello World!" Plugin**
+
+- When the plugin is generated, the CLI scaffolds several files in the plugins/my-plugin directory.
+- To create a simple "Hello World" plugin, strip down the contents of `\plugins\simple-plugin\src\components\ExampleComponent\ExampleComponent.tsx` to the following:
+
+```tsx
+import React from 'react';
+import { Content, Page } from '@backstage/core-components';
+
+export const ExampleComponent = () => (
+  <Page themeId="tool">
+    <Content>
+      <h1>Hello, World!</h1>
+      <p>This is your simple plugin.</p>
+    </Content>
+  </Page>
+);
+```
+- As the component has been significantly simplified, you can remove unnecessary files such as the test file `ExampleComponent.test.tsx` and the entire    `ExampleFetchComponent` folder.
+- Your plugin has been reduced to a minimal "Hello World!" setup and is ready for further development.
+
+**Step 3: Integrate into the Software Catalog**
+
+1. Remove the following lines from `packages/app/src/App.tsx` to avoid direct routing:
+
+```tsx
+import { SimplePluginPage } from '@internal/backstage-plugin-simple-plugin';
+...
+<Route path="/simple-plugin" element={<SimplePluginPage />}/>
+```
+2. Add the following lines to `packages/app/src/components/catalog/EntityPage.tsx` to include your plugin as tabbed content:
+
+```tsx
+import { SimplePluginPage } from '@internal/backstage-plugin-simple-plugin';
+...
+const serviceEntityPage = (
+  <EntityLayout>
+    ...
+    <EntityLayout.Route path="/simple-plugin" title="My Plugin">
+      <SimplePluginPage />
+    </EntityLayout.Route>
+  </EntityLayout>
+);
+```
+3. Rebuild or restart the Backstage App image.
+4. Create a new GitHub repository `(e.g., my-simple-plugin-component)` with a `catalog-info.yaml` file:
+```yaml
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: my-simple-plugin-component
+spec:
+  lifecycle: production
+  owner: my-team
+  type: service
+```
+
+5. Use the `HOME > CREATE > REGISTER EXISTING COMPONENT` button to register your service component using the link to the `catalog-info.yaml` file.
+   
+![backstage-doc-2](https://github.com/user-attachments/assets/8a36ca0a-f2b4-4626-a6ed-8ab754f1bb66)
```

Comment:
I can't see any of the images

Reply:
sorry to hear that, i will make sure to add images in a new folder of repo and refer them here.